### PR TITLE
Fix typos in DefaultTabController.of documentation

### PR DESCRIPTION
**Pull request by ThinkDigitalSoftware**
Fix error in `DefaultTabController.of` documentation

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
- [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
- [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
- [ ] I wrote a migration guide: *Replace with a link to your migration guide*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by ThinkDigitalSoftware** [comment]
@Hixie It was just a documentation change, so no new tests were needed

**Review by shihaohong — DISMISSED** [review]
LGTM!

**Comment by Hixie** [comment]
Actually this does need a test. Or more to the point, it needs to use the syntax that triggers the analyzer on this code, such that the error can't happen in the first place. I forget exactly what the syntax is, but if you look at other sample code it should be clear.

**Comment by ThinkDigitalSoftware** [comment]
> Actually this does need a test. Or more to the point, it needs to use the syntax that triggers the analyzer on this code, such that the error can't happen in the first place. I forget exactly what the syntax is, but if you look at other sample code it should be clear.

If you're talking to me, This is well above my paygrade, lol. I didn't know comments could be tested.

**Comment by shihaohong** [comment]
Here's a [pointer](https://github.com/flutter/flutter/blob/master/dev/bots/analyze-sample-code.dart) to how it's done in flutter/flutter, but one way you can do it is wrap it in `{@tool snippet ...}{@end-tool}`

**Comment by ThinkDigitalSoftware** [comment]
@shihaohong  did I do it right?

**Comment by Hixie** [comment]
I think you'll also have to add the following to the top of the file:
```
// Examples can assume:
// BuildContext context;
```
I'm working on documenting this better. Our docs are stale here.

**Comment by ThinkDigitalSoftware** [comment]
According to the docs, it says that that's for specific cases. 
https://github.com/flutter/flutter/blob/fc9b30a4f1d572f80d140b9f1ef5bbd3a0152407/dev/bots/analyze-sample-code.dart#L32-L36

does that apply here?

**Comment by shihaohong** [comment]
It applies here because `context` is not defined in your dart snippet

File: packages/flutter/lib/src/material/tab_controller.dart [hunk 1]
```diff
@@ -333,14 +333,16 @@ class DefaultTabController extends StatefulWidget {
   ///
   /// {@macro flutter.widgets.child}
   final Widget child;
-
+
```
Comment (shihaohong) [inline packages/flutter/lib/src/material/tab_controller.dart]: I would reformat it in the following manner:

```
  /// The closest instance of this class that encloses the given context.
  ///
  /// {@tool sample}
  /// Typical usage is as follows: 
  /// ```dart
  /// TabController controller = DefaultTabController.of(context);
  /// ```
  /// {@end-tool}
```

Comment (shihaohong) [reply packages/flutter/lib/src/material/tab_controller.dart]: That way, the first sentence of the API doc will not be part of the code sample.

**Comment by shihaohong** [comment]
Hi, do you happen to have any updates on working through this PR?

**Comment by ThinkDigitalSoftware** [comment]
I haven't gotten back to it. Sorry. I don't mind if you add it, but I've
gotten to busy to work on code these past few days

**Review by justinmc — APPROVED** [review]
LGTM! 👍

**Pull request merged**